### ISSUE TITLE
Can't read achievement notifications in race

### ISSUE THREAD

**Issue by hiker** [supertuxkart/stk-code#2068]
That was on osx, 0.9rc1 - I only see the box, no text. We had that before #1940 - not sure if it's the same or not; if it's only osx or others as well.

**Comment by hiker** [comment]
On windows, hd3000, 0.9rc1 it works.

**Comment by hiker** [comment]
Update: looks like my bug fix was removed in 06c8089d. Still, the code works on VS, but not osx, @auriamg to investigate why it doesn't work on osx. We might to revert more removed c_str() functions if this should be an osx issue.

**Comment by Flakebi** [comment]
I did some testing on linux with clang (version 3.6.0), with gcc it works. Adding `.c_str()` doesn't fix it for me (though I don't know about osx). I went back to  dd7eb2c (the last commit that is easy to compile for me) and I can't see the achievement messages either.

The only way that worked so far is like in #1960:
```
     if(m_achievement_info->checkCompletion(this))
     {
         //show achievement
+        const core::stringw &name = m_achievement_info->getName();
         core::stringw s = _("Completed achievement \"%s\".", 
-                            m_achievement_info->getName());
+                            name);
         MessageQueue::add(MessageQueue::MT_ACHIEVEMENT, s);
```
